### ISSUE TITLE
Add an interface for http.Client

### ISSUE THREAD

**Issue by Dynom** [golang/go#16047]
**TL;DR** I'm trying to add some robustness to web service requests, but I find it very difficult to add this without an interface for `http.Client`.

*To sketch the situation:*
* [This](https://github.com/google/google-api-go-client/blob/master/translate/v2/translate-gen.go#L48) factory (google-api-go-client Translate) requires an `http.Client` as argument.
* [This](https://github.com/sethgrid/pester) library wraps `http.Client` to provide all sorts of great features for adding robustness. 
* Because Go doesn't offer an interface type for `http.Client`, only concrete types can be used and thus there is no (sensible) way to use `pester.Client` even though it is a drop-in replacement for `http.Client`.

*What I've done so far*
* I've searched for an interface that can be used by the Translate SDK so that it can be changed to accept a custom concrete type, but I didn't find any.

A related thread: https://github.com/google/google-api-go-client/issues/142#issuecomment-225371632

**Comment by ianlancetaylor** [comment]
The place to fix this is google-api-go-client.

Adding an interface for `http.Client` in the http package does not seem right.  That suggests that other people will want to implement the interface.  Since they will do, it will be impossible for us in practice to add new methods to the `http.Client` interface, since that would break other packages.

Instead, packages that want to use `http.Client`, and want to permit wrapping, should define their own interface, and use it.

Closing because I don't think there is anything to do in the Go standard library.

**Comment by Dynom** [comment]
Alright, thanks for the explanation.
